Mathlib/RingTheory/MvPolynomial/Localization.lean

```diff
@@ -69,3 +79,78 @@ lemma isLocalization_C_mk' (a : R) (m : M) :
     IsLocalization.mk'_spec]
 
 end MvPolynomial
+
+namespace IsLocalization.Away
+
+open MvPolynomial
+
+variable (r : R) [IsLocalization.Away r S]
+
+/-- The canonical algebra map from `MvPolynomial Unit R` quotiented by
+`C r * X () - 1` to the localization of `R` away from `r`. -/
+private noncomputable
+def auxHom : (MvPolynomial Unit R) ⧸ (Ideal.span { C r * X () - 1 }) →ₐ[R] S :=
+  Ideal.Quotient.liftₐ (Ideal.span { C r * X () - 1}) (aeval (fun _ ↦ invSelf r)) <| by
+    intro p hp
+    refine Submodule.span_induction hp ?_ ?_ ?_ ?_
+    · rintro p ⟨q, rfl⟩
+      simp
+    · simp
+    · intro p q hp hq
+      simp [hp, hq]
+    · intro a x hx
+      simp [hx]
+
+@[simp]
+lemma auxHom_mk (p : MvPolynomial Unit R) :
```

Comment:
Should this also be `private`? It uses a private definition in the statement, which will then not be understandable outside this file.